### ISSUE TITLE
Ignore HTTP Accept header in GET methods

### ISSUE THREAD

**Issue by marcelel** [micronaut-projects/micronaut-core#6621]
### Issue description
Hello. I would like to send a response with `application/json` regardless of request’s `Accept`  header. It was working out of the box on the previous versions of Micronaut, but is not after upgrading to the latest version. The only way I was able to do it was to implement `HttpFilter` which adds a `Content-Type` header to each response.
```java
@Filter("/**")
public class ContentTypeHeaderFilter implements HttpFilter {

    @Override
    public Publisher<? extends HttpResponse<?>> doFilter(HttpRequest<?> request, FilterChain chain) {
        return Flowable.fromPublisher(chain.proceed(request))
                .map(this::replaceContentTypeHeader);
    }

    private HttpResponse<?> replaceContentTypeHeader(HttpResponse<?> response) {
        ((MutableHttpHeaders) response.getHeaders()).set(HttpHeaders.CONTENT_TYPE, APPLICATION_JSON);
        return response;
    }
}
```

### Expected behaviour 
Send a response with `application/json` `Content-Type` header regardless of GET request’s `Accept` header

### Actual behaviour 
Send a `406` HTTP response

### Steps to reproduce

1. Create an endpoint:
```java
@Get(value = "/test", produces = MediaType.APPLICATION_JSON)
```
2. Send a request with `Accept` header with a value other than `application/json`, e.g. `application/xml`

### Version
Micronaut 3.1.3

**Comment by yawkat** [comment]
The content negotiation support can be improved. I will work on that.

However, I think this particular case is fine. If the client sends an accept header that is incompatible with any route, then [the spec](https://httpwg.org/specs/rfc7231.html#header.accept) says:

> If the header field is present in a request and none of the available representations for the response have a media type that is listed as acceptable, the origin server can either honor the header field by sending a 406 (Not Acceptable) response or disregard the header field by treating the response as if it is not subject to content negotiation.

We do the former as far as I can tell, which sounds sensible.

If you want to match any request, you could also set `produces = "*/*"` and set the `Content-Type` in a `HttpResponse` you return from the controller. It doesn't need a filter.

**Comment by yawkat** [comment]
Apparently it's also possible to do something like `@Get(produces = {'application/json', '*/*'})`

**Comment by marcelel** [comment]
Thank you for the response. I don't consider it as a bug as well. I compared it with a Spring Boot application and it behaves in the same way. On the other hand, [other spec](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/406) also says:

> In practice, this error is very rarely used. Instead of responding using this error code, which would be cryptic for the end user and difficult to fix, servers ignore the relevant header and serve an actual page to the user. It is assumed that even if the user won't be completely happy, they will prefer this to an error code.

IMHO it can be a nice feature to be able to have the default response in the most common format (such as `application/json`) instead of returning 406.

Once again, I don't find this a bug (sorry - should highlight it at the beginning). I added this `ContentTypeHeaderFilter` and marked each controller with `produces = MediaType.ALL` - works fine.
